src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java

```diff
@@ -44,6 +45,35 @@
   private ProviderContext providerContext;
   private Integer providerContextIndex;
 
+  private static final Map<Class<?>, Class<?>> primitiveBoxingMap =
+    new HashMap<Class<?>, Class<?>>() {
+      { put(byte.class,    Byte.class); }
+      { put(short.class,   Short.class); }
+      { put(int.class,     Integer.class); }
+      { put(long.class,    Long.class); }
+      { put(float.class,   Float.class); }
+      { put(double.class,  Double.class); }
+      { put(boolean.class, Boolean.class); }
+      { put(char.class,    Character.class); }
+    };
+
+  private static boolean isAssignableFrom(Class<?> to, Class<?> from) {
+    if (to.isAssignableFrom(from)) {
+      return true;
+    }
+    if (to.isPrimitive()) {
+      return isBoxingType(from, to);
+    }
+    return false;
+  }
+
+  private static boolean isBoxingType(Class<?> target, Class<?> primitive) {
+    if (!primitive.isPrimitive()) {
```

Comment:
This code is dead-code because same check is performed on caller. I will suggest to remove this method and modify `isAssignableFrom` method as follow:

```java
  private static boolean isAssignableFrom(Class<?> to, Class<?> from) {
    if (to.isAssignableFrom(from)) {
      return true;
    }
    if (to.isPrimitive()) {
      return from == primitiveBoxingMap.get(to);
    }
    return false;
  }
```